force-app/main/default/classes/DMLOperations.cls

```diff
@@ -97,10 +104,11 @@ public with sharing class DMLOperations {
     public static void updateOpportunityStage(Id oppId, String newStage) {
         // Retrieve the Opportunity from the database
         // [SELECT Id FROM Opportunity WHERE Id = :oppId]
-
+        Opportunity oppToUpdate = [SELECT Id FROM OPPORTUNITY WHERE ID = :oppId];
```

Comment:
I would also query for StageName. Not 100% necessary, but good practice in case you wanted to validate the prior value before updating.